internal/splunkconfig/config/indexarchiver.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import "fmt"
+
+// Archiver represents the value of an Index's "archiver.coldStorageProvider" fields.
+type IndexArchiver string
```

Comment:
This is the IndexArchiverProvider is it not?

Reply:
According to index.conf https://docs.splunk.com/Documentation/Splunk/9.1.2/Admin/Indexesconf It is called `coldStorageProvider`

Reply:
I don't much about `golang`, but should we just call it `DynamicDataActiveArchiver` so it's clear and it's what Splunk Cloud calls it?

Reply:
I was trying to avoid using something as verbose as `IndexArchiverColdStorageProvider` but we could certainly do that to avoid any further confusion. I was just suggesting we be more descriptive than `IndexArchiver` which is more than the `coldStorageProvider`. Let me know if that makes more sense, otherwise happy to chat offline.

Reply:
are you guys ok if i rename it to `coldStorageProvider`

Reply:
So that's a specific option within the DDAA config in indexes.conf - the changes we are making to this MR is to add DDAA options to the config. Hence why `DynamicDataActiveArchiver` makes sense since it clearly states what these configs are and not point to a specific DDAA config.

Reply:
I think there's some confusion about what this type is doing in the provider. This is defining a string that can only have specific values which corresponds to the cold provider, hence the name suggestion that I made.

Reply:
Sorry about the confusion, totally missed that since GH doesn't allow you to collapse a comment thread to see what's being talked about -sigh-